### ISSUE TITLE
Display of error bars on log scale

### ISSUE THREAD

**Issue by hoogerheide** [reductus/reductus#111]
When plotting data where the signal is comparable to or smaller than the background, two situations frequently arise that are perfectly normal:
1. Most commonly, one of the error bars of the background-subtracted data point goes below zero. In this case the error bar is not currently plotted.
2. The background-subtracted data point itself is less than zero. In this case the point appears to be missing completely.

In case 2, at least 2/3 of the time the 68% CI should include zero, i.e. the top of the error bar should still be plottable on a log scale. It would be extremely useful to see this error bar when plotting on a log scale. Right now it can look like chunks of data are missing even when statistically they're consistent with zero. This has led to unwarranted angst among users about the quality of their data.

My suggestion is to always show the error bars, even if they dip below zero (case 1), or the top of the error bar even if the data point is below zero (case 2). To avoid unhelpful effects with axis scaling, I would suggest that axis scaling be done ignoring the error bars and the error bars plotted later or cut off at the bottom of the (already scaled) plot.

As an example, see this linear plot, which shows perfectly fine background-subtracted data around Q = 0.3:

![image](https://user-images.githubusercontent.com/14282356/102714834-6d8b2180-429f-11eb-9077-6812a178a0d9.png)

And this log plot, which looks like all the data there are oversubtracted:

![image](https://user-images.githubusercontent.com/14282356/102714851-90b5d100-429f-11eb-948a-5ce6b00bc871.png)

**Comment by bmaranville** [comment]
Yes - this is a little bit tricky, but probably doable.  The issue is that there be infinities here... and they have to be trapped and converted to the nearest appropriate pixel.
